app/services/create_new_device_alert.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class CreateNewDeviceAlert < ApplicationJob
+  include UserAlerts
+  queue_as :long_running
+
+  def perform(now)
+    emails_sent = 0
+    User.where(
+      sql_query_for_users_with_new_device,
+      tvalue: now - IdentityConfig.store.new_device_alert_delay_in_minutes.minutes,
+    ).each do |user|
+      emails_sent += 1 if clear_new_device_and_send_email(user)
+    end
+
+    emails_sent
+  end
+
+  private
+
+  def sql_query_for_users_with_new_device
+    <<~SQL
+      sign_in_new_device_at < :tvalue AND
+      sign_in_new_device_at IS NOT NULL
```

Comment:
I might recommend swapping these two conditions? Or maybe even getting rid of `IS NOT NULL` check if this is accounted for already with the `<` check?

The thinking being that a `NULL` check might be quicker to filter on and prevent any ambiguity around how `<` treats `NULL` values (related to my previous comment).

```suggestion
      sign_in_new_device_at IS NOT NULL AND
      sign_in_new_device_at < :tvalue
```